src/ubx.cpp

```diff
@@ -533,16 +536,17 @@ int GPSDriverUBX::configureDevice(const GNSSSystemsMask &gnssSystems)
 	// measurement rate
 	// In case of F9P not in moving base mode we use 10Hz, otherwise 8Hz (receivers such as M9N can go higher as well, but
 	// the number of used satellites will be restricted to 16. Not mentioned in datasheet)
-	int rate_meas;
-
-	if (_mode != UBXMode::Normal) {
-		rate_meas = 125; //8Hz for heading.
+	int rate_meas_ms = 125;
+	int heading_rate_hz = 8;
 
-	} else {
-		rate_meas = (_board == Board::u_blox9_F9P) ? 100 : 125;
+	if (_board == Board::u_blox9_F9P && (_mode == UBXMode::Normal || gnssSystems != GNSSSystemsMask::RECEIVER_DEFAULTS)) {
+		// (U-Blox F9P only)
+		// If not calculating heading, or if specifying specific constellations, then use a 10Hz update rate
+		rate_meas_ms = 100;
+		heading_rate_hz = 10;
 	}
```

Comment:
@benflyvoly this is equivalent to what you had before, but I wasn't 100% sure if the original version was correct or not.  Worth checking.